### PR TITLE
prevent nested prop values from being overwritten

### PR DESCRIPTION
**Pull request by natebrennand**
**Overview:**
Depending on the order of props provided to a component that leverages nested components, a nested property may be overwritten.

**Testing:**
- [x] Unit tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "clever-components",
-  "version": "0.20.0",
```
Comment (cozmo) [inline package.json]: This could be a patch I feel like?

**Comment by natebrennand** [comment]
CC @osdiab

File: package.json (hunk 1 above)
Comment (natebrennand) [inline package.json]: Added a behavior change w/ https://github.com/Clever/components/pull/139/commits/8e595f2fa8205c27c2f915d3f2f77e3e78b49597 which I think warrants minor

File: src/ConfirmationButton/ConfirmationButton.jsx [hunk 2]
```diff
@@ -35,16 +37,17 @@ export class ConfirmationButton extends React.Component {
...
+  {
     onConfirm: PropTypes.func,
-  }
+  },
 );
 
-ConfirmationButton.defaultProps = Object.assign(
-  {}, prefixKeys(Button.defaultProps, "confirmButton"), ModalButton.defaultProps, {
+ConfirmationButton.defaultProps = Object.assign({},
+  prefixKeys(Button.defaultProps, propPrefix), ModalButton.defaultProps,
```
Comment (amir-clever) [inline src/ConfirmationButton/ConfirmationButton.jsx]: Nate' requests Nate' to add a newline

**Pull request merged**